orchprovider/k8s/v1/util.go

```diff
@@ -193,6 +193,74 @@ func (p *VolumeMarkerBuilder) AddControllerStatuses(pod k8sApiV1.Pod) {
 	_ = p.AddMultiples(string(v1.JivaControllerStatusVK), status, true)
 }
 
+// AddControllerContainerStatuses is to fetch the current state of controller containers
+// inside the pod
+func (p *VolumeMarkerBuilder) AddControllerContainerStatuses(cp k8sApiV1.Pod) {
+	for _, current := range cp.Status.ContainerStatuses {
+		value := v1.NilVV
+		if current.State.Waiting != nil {
+			value = v1.ContainerWaitingVV
+		}
+
+		if current.State.Terminated != nil {
+			value = v1.ContainerTerminatedVV
+		}
+
+		if current.State.Running != nil {
+			if current.Ready == true {
+				value = v1.ContainerRunningVV
+			} else {
+				value = v1.ContainerNotRunningVV
+			}
+		}
+		_ = p.AddMultiples(string(v1.ControllerContainerStatusVK), string(value), true)
+	}
+}
+
+// AddReplicaContainerStatuses is to fetch the current state of replica containers
+// inside the pod
+func (p *VolumeMarkerBuilder) AddReplicaContainerStatuses(cp k8sApiV1.Pod) {
+	for _, current := range cp.Status.ContainerStatuses {
```

Comment:
Have a new method called AddContainerStatuses that does all these.
This will be called from AddReplicaContainerStatuses & AddControllerContainerStatuses by passing specific KEYs i.e. ReplicaContainerStatusVK & ControllerContainerStatusVK

Overall abstract the common stuff. This two methods are redundant.

Reply:
done